### ISSUE TITLE
focus/keyboard input issues in preview window

### ISSUE THREAD

**Issue by aronatkins** [rstudio/rstudio#2301]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.2.375
    OS Version      : macOS 10.13.3
    R Version       : 3.4.3

### Steps to reproduce the problem

Given the R Markdown document:
````
---
output: revealjs::revealjs_presentation
---

## First!

here is the first slide

## Aww. Second.

this is not the first slide
````

Knit the document (with Command-Shift-K) and have it open in a preview window.

Knit settings:
<img width="254" alt="screen shot 2018-03-02 at 9 12 52 am" src="https://user-images.githubusercontent.com/362187/36902866-f141a5ee-1df9-11e8-9adc-73bf72003567.png">

### Describe the problem in detail

The resulting preview window initially does not respond to arrow events (left, right). If you click in the window to give it focus, it then responds to left/right events.

The preview window does not respond to Command-W events even after click-to-focus.

### Describe the behavior you expected

* The preview window responds to left/right events immediately after opening (it does not require click-to-focus).
* The preview window is always closed by Command-W.

```
> sessionInfo()
R version 3.4.3 (2017-11-30)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.3

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.4/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods  
[7] base     

loaded via a namespace (and not attached):
 [1] compiler_3.4.3  backports_1.1.2 magrittr_1.5    rsconnect_0.8.7
 [5] rprojroot_1.3-2 htmltools_0.3.6 tools_3.4.3     revealjs_0.9   
 [9] yaml_2.1.16     Rcpp_0.12.14    stringi_1.1.6   rmarkdown_1.9  
[13] knitr_1.20      stringr_1.2.0   digest_0.6.15   evaluate_0.10.1
```

**Comment by aronatkins** [comment]
The Command-W problem is not specific to R Markdown previews - it is also visible with simple Shiny applications opened in the preview window.

**Comment by aronatkins** [comment]
Maybe similar: A popped-out source-editing window responds to Command-W but only after a mouse-click in the window.

**Comment by aronatkins** [comment]
Maybe similar: A git "diff" window that pops up does not respond to Command-W.

**Comment by jmcphers** [comment]
Lots of weird happening here post `QtWebEngine` change. I'm investigating.

**Comment by jmcphers** [comment]
This is the code that *used* to make this work:

https://github.com/rstudio/rstudio/blob/42b53e081174f0ca1ae65f353f9df0376283690a/src/cpp/desktop-mac/MainFrameController.mm#L346-L357

in other words, we'd intercept Cmd+W and, if the `closeSourceDoc` command was not enabled, we'd close the window instead. No such logic current exists on the `QtWebEngine` side so probably needs to be ported there.

**Comment by dfalty** [comment]
Verified fixed on version 1.2.1090.
